x-pack/legacy/plugins/file_upload/public/util/http_service.js

```diff
@@ -40,38 +36,22 @@ export async function http(options) {
   if (body !== null) {
     payload.body = body;
   }
-  return await fetchWithTimeout(url, payload);
+  return await doFetch(url, payload);
 }
 
-async function fetchWithTimeout(url, payload) {
-  let timedOut = false;
-
-  return new Promise(function (resolve, reject) {
-    const timeout = setTimeout(function () {
-      timedOut = true;
-      reject(new Error(
-        i18n.translate('xpack.fileUpload.httpService.requestTimedOut',
-          { defaultMessage: 'Request timed out' }))
-      );
-    }, FETCH_TIMEOUT);
-
-    fetch(url, payload)
-      .then(resp => {
-        clearTimeout(timeout);
-        if (!timedOut) {
-          resolve(resp);
-        }
-      })
-      .catch(function (err) {
-        reject(err);
-        if (timedOut) return;
-      });
-  }).then(resp => resp.json())
-    .catch(function (err) {
-      console.error(
+async function doFetch(url, payload) {
```

Comment:
no need for the tmp vars and the if-then-else to check for success. the try catch already does the branching.

Consider simplifying
```
  try {
    const resp = await fetch(url, payload);
    return resp.json();
  } catch(err) {
    return {
      failures: [
        i18n.translate('xpack.fileUpload.httpService.fetchError', {
          defaultMessage: 'Error performing fetch: {error}',
          values: { error: err.message }
        })]
    };
  }
```